### PR TITLE
Better docs, including tutorial

### PR DESCRIPTION
**Pull request by n1k0**
Refs #10 

- [x] Migrate docs to mkdocs
- [x] Draft tutorial
- [x] Update current demo code to use the tutorial result one
- [ ] Set up RTD integration

### CODE REVIEW

File: docs/tutorial.md [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+```
+
+**Note:** You'll need to serve this page over HTTP, for Cliquetis to work. To do so, you can use node's [http-server](https://github.com/indexzero/http-server), Python's [SimpleHTTPServer](https://docs.python.org/2/library/simplehttpserver.html) or whatever Web server you like.
+
+For example, if you're using http-server:
+
+    $ npm install -g http-server
+    $ http-server yourdirectory
+
+And that's it. You should see something like this:
```
Comment (leplatrem) [inline docs/tutorial.md]: like this on ``http://localhost:XXXX``:

File: docs/tutorial.md [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+  var db = new Cliquetis();
+  var tasks = db.collection("tasks");
+
+  document.getElementById("form")
+    .addEventListener("submit", function(event) {
+      event.preventDefault();
+      tasks.create({
+        label: event.target.label.value,
+        done: false
+      }).then(res => {
```
Comment (leplatrem) [inline docs/tutorial.md]: Isn't better indented like this ?
(code layout is like the hardest part of Promises :))

```
    tasks.create({
        label: event.target.label.value,
        done: false
      })
      .then(res => {
```

File: docs/tutorial.md [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+Though if you're using Firefox, open your [Developer Tools]() and head to the *Storage* tab; you should see some IndexedDB databases listed there, and should find you're recently created tasks:
+
+![](images/indexeddb-inspector.png)
+
+Now, refresh the page; your data are still there, thank you [indexedDB](https://developer.mozilla.org/fr/docs/IndexedDB)!
+
+Last, switch off your Internet connection, and try adding tasks. It still works, because we're offline first, remember. We'll get through synchronization a bit later.
+
+- XXX: a note on Promises
+- XXX: a note on result format (res.data)
```
Comment (leplatrem) [inline docs/tutorial.md]: XXX a note on ES6 ?

**Comment by n1k0** [comment]
This is now ready for an early review. I suggest reviewers to take the time to actually try the tutorial, as it's super important it's up to date and accurate for newcomers.

r=? @leplatrem @Natim @ametaireau

**Comment by n1k0** [comment]
Current status: https://twitter.com/n1k0/status/611868022039543808

**Pull request merged**